executor/join.go

```diff
@@ -643,30 +658,32 @@ func (e *HashJoinExec) Next(ctx context.Context, req *chunk.Chunk) (err error) {
 	return nil
 }
 
-func (e *HashJoinExec) handleFetchAndBuildHashTablePanic(r interface{}) {
+func (e *HashJoinExec) handleFetchBuildSidePanic(r interface{}) {
 	if r != nil {
-		e.buildFinished <- errors.Errorf("%v", r)
+		e.fetchBuildSideFinished <- errors.Errorf("%v", r)
 	}
-	close(e.buildFinished)
+	close(e.fetchBuildSideFinished)
+}
+
+func (e *HashJoinExec) handleBuildHashTablePanic(r interface{}) {
+	if r != nil {
+		e.buildHashtableFinished <- errors.Errorf("%v", r)
+	}
+	close(e.buildHashtableFinished)
 }
 
 func (e *HashJoinExec) fetchAndBuildHashTable(ctx context.Context) {
 	// buildSideResultCh transfers build side chunk from build side fetch to build hash table.
 	buildSideResultCh := make(chan *chunk.Chunk, 1)
 	doneCh := make(chan struct{})
-	go util.WithRecovery(func() { e.fetchBuildSideRows(ctx, buildSideResultCh, doneCh) }, nil)
+	go util.WithRecovery(func() { e.fetchBuildSideRows(ctx, buildSideResultCh, doneCh) }, e.handleFetchBuildSidePanic)
 
 	// TODO: Parallel build hash table. Currently not support because `rowHashMap` is not thread-safe.
 	err := e.buildHashTableForList(buildSideResultCh)
 	if err != nil {
-		e.buildFinished <- errors.Trace(err)
+		e.buildHashtableFinished <- err
 		close(doneCh)
 	}
```

Comment:
We can create the temparory variable `fetchBuildSideFinished` (not nessacerray to be a entry of `HashJoinExec`) before `go e.fetchBuildSideRows` and check the result of this channel immediately after `buildHashTableForList` finished.

Reply:
But why do we need this variable?